mlir_graphblas/src/lib/GraphBLAS/GraphBLASLowerPass.cpp

```diff
@@ -652,8 +674,14 @@ class LowerSelectRewrite : public OpRewritePattern<graphblas::SelectOp> {
     Type memref1DI64Type = MemRefType::get({-1}, int64Type);
     Type memref1DValueType = MemRefType::get({-1}, valueType);
 
+    std::string selector = op.selector().str();
     OperandRange thunks = op.thunks();
-    ArrayAttr selectors = op.selectors();
+    llvm::Optional<Value> thunk = llvm::None;
+    if (thunks.size() > 0)
+      thunk = thunks[0];
+    llvm::Optional<Value> rngContext = llvm::None;
+    if (thunks.size() > 1)
```

Comment:
We can assume that the number of thunks is either `0`, `1`, or `2` because of verification, but it's nice to make this assumption explicit and self-documenting in this code here by having the checks be `thunks.size() == 0`, `thunks.size() == 1`, etc. to reduce the possibility that a bug be introduced by someone who isn't aware of this assumption. A newer developer might mistakenly assume that the number of thunks could be 3. 

I don't have strong feelings about this. I'd be fine leaving it as is, but this idea is worth considering.

Reply:
Good idea. Clarity of guaranteed state is always helpful when checks are far from the logical code.